### PR TITLE
add home assignment 2

### PR DESCRIPTION
**Pull request by zeevso**

### CODE REVIEW

File: home-assignments/session1/weather.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                      params={'q': location, 'units': 'metric', 'APPID': 'a08359a0cfaf6172ada114041887b2f0'}).json()
+    return r
+
+
+def get_country_by_code(country_code, countries):
+    for country in countries:
+        if country['code'] == country_code:
+            country_name = country['name']
+            return (country_name)
+    return None
```
Comment (li0nbelenky) [inline home-assignments/session1/weather.py]: 'return None' statement is redundant in python.
The function will return None anyway.
image

File: home-assignments/session1/weather.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+import json
+import requests
+import countryinfo
+
+OUTPUT_FILE = 'weather.txt'
+
+def get_public_ip():
+    r = requests.get("http://jsonip.com/").json()
+    return r["ip"]
```
Comment (li0nbelenky) [inline home-assignments/session1/weather.py]: Protect your IO functions with try/except in cases of bad connectivity issues(http errors 400/500 or socket issues) or an incorrect JSON structure should return.

File: home-assignments/session1/exercise2.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        return self.min_age <= age < self.max_age
+
+
+def read_json():
+    with open(INPUT_FILE) as file:
+        return json.load(file)
+
+
+def init_data(data):
+    data = read_json()
```
Comment (li0nbelenky) [inline home-assignments/session1/exercise2.py]: data is unused, you pass the output from read_json() from your main function,
and then re-assign it here again. so you can either remove it from main, or from here :)

Comment (zeevso) [reply home-assignments/session1/exercise2.py]: OK. Thanks

**Pull request closed without merging**